app/models/user.rb

```diff
@@ -1,16 +1,47 @@
 class User < ApplicationRecord
-  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]|\z/i
+  attr_accessor :remember_token
+  has_one :cart
+
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
   before_save { email.downcase! }
-  validates :name, presence: true, length: { maximum: 50 }
-  validates :email, presence: true, length: { maximum: 255, minimum: 6 },
-            format: { with: VALID_EMAIL_REGEX },
+  validates :name, presence: true, length: {maximum: 50}
+  validates :email, presence: true, length: {maximum: 255, minimum: 6},
```

Comment:
rails has a nice already implemented regex for emails:
```ruby
validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }, presence: true, uniqueness: true
```

Reply:
The regex was an exercise, part of the tutorial. I will replace it with your version, I can agree it looks better.